### PR TITLE
Adding `--version` flag to microk8s

### PR DESCRIPTION
**Pull request by lferran**
## Goal
The purpose of this ticket is to add the `--version` flag to the main `microk8s` entrypoint. It will also work when invoked with `microk8s version`, to be conformant with related tools like `microk8s ctr version`, `docker version` or `kubectl version`.

## Motivation
As of now, we are using `snap info microk8s` and `microk8s.kubectl version` to know which versions of the snap and upstream k8s are installed, correspondingly. We are missing a convenient and unified way to get version information.

We want to show the version for the following components:
- [x] MicroK8s snap
- [x] Upstream k8s
- [x] cni

## Example output
  ```bash
 $ microk8s version      
MicroK8s v1.21.11 revision 3058
 ```

### CODE REVIEW

**Comment by lferran** [comment]
It seems that writing the versions.json file is not allowed
```bash
Writing versions at /snap/snapcraft/7201/versions.json ...
/bin/sh: 9: /root/parts/runc/src/set-env-variables.sh: cannot create /snap/snapcraft/7201/versions.json: Read-only file system
+ echo {"kube":"v1.23.5","cni":"v0.8.7"}
```
Alternatively, if there is a way to get the cni version from python, we could drop the file approach.

**Comment by lferran** [comment]
I managed to install the snap locally and test it:
```bash
$ microk8s --version
MicroK8s v1.23.6 revision: x1
  - K8s: v1.23.6
  - CNI: v0.8.7
```
and also
```bash
$ sudo snap run --shell microk8s
root@pardal:~# cat $SNAP/versions.json
{"kube":"v1.23.6","cni":"v0.8.7"}
```

**Comment by glebsts** [comment]
@lferran though you did a great job, thank you, mb there is a point to align this feature with other "get version" commands like `microk8s ctr version`, `docker version`, `kubectl version` etc, and make it `version`, not `--version`?

**Comment by barrettj12** [comment]
> @lferran though you did a great job, thank you, mb there is a point to align this feature with other "get version" commands like `microk8s ctr version`, `docker version`, `kubectl version` etc, and make it `version`, not `--version`?

Why not support both?

File: microk8s-resources/wrappers/microk8s.wrapper [hunk 1]
```diff
@@ -27,6 +27,9 @@ elif [ "${APP}" == "inspect" ]; then
 elif [ "${APP}" == "help" ] || [ "${APP}" == "--help" ] || [ "$APP" == "-h" ]; then
     help
     readonly EXIT="0"
+elif [ "${APP}" == "--version" ] || [ "${APP}" == "-v" ]; then
+    "${SNAP}/microk8s-version"
```
Comment (neoaggelos) [inline microk8s-resources/wrappers/microk8s.wrapper:31]: If we changed the name of the script to `microk8s-version.wrapper`, we would basically also get the `microk8s version` flavor of the command for free.

File: scripts/wrappers/version.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+def get_upstream_versions() -> str:
+    versions_file = Path(environ["SNAP"]) / "versions.json"
+    with open(versions_file, mode="r") as file:
+        versions = json.loads(file.read())
+        return versions
+
+
+def get_snap_version() -> str:
+    return environ["SNAP_VERSION"]
```
Comment (neoaggelos) [inline scripts/wrappers/version.py:16]:
```suggestion
    return os.getenv("SNAP_VERSION", "(unknown)")
```

File: scripts/wrappers/version.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def get_snap_revision() -> str:
+    return environ["SNAP_REVISION"]
```
Comment (neoaggelos) [inline scripts/wrappers/version.py:20]: Similarly here

File: tests/unit/test_version.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+
+@patch("version.environ", {"SNAP_VERSION": "v1.21.11"})
+def test_get_snap_version():
+    assert get_snap_version() == "v1.21.11"
+
+
+@patch("version.environ", {"SNAP_REVISION": "3086"})
+def test_get_snap_revision():
+    assert get_snap_revision() == "3086"
```
Comment (neoaggelos) [inline tests/unit/test_version.py:19]: I would probably avoid mocking for this, and instead just set the environment variable

**Review by neoaggelos — COMMENTED** [review]
Some minor comments

File: scripts/wrappers/version.py (hunk 2 above)
Comment (lferran) [reply scripts/wrappers/version.py:16]: isn't `SNAP_VERSION` a variable that we expect to be there always?

File: tests/unit/test_version.py (hunk 4 above)
Comment (lferran) [reply tests/unit/test_version.py:19]: sure! Why though? Out of curiosity

Comment (neoaggelos) [reply tests/unit/test_version.py:21]: This way, the tests are too closely tied to the implementation (e.g. if the implementation were to use `os.getenv` instead, these tests would not work). 

Mocks are best suited for outputs/integration with external services. They also serve as a way of disconnecting the testing environment from the real one. This is useful when we want to mock the output of an external command, etc, but is probably a dis-service when checking environment (where no mocks are really needed, the test case itself is isolated).

Comment (lferran) [reply tests/unit/test_version.py:21]: makes total sense! thanks for sharing your thoughts

**Review by ktsakalozos — APPROVED** [review]
LGTM +1 thank you @lferran

**Pull request merged**